source/game/field/KCollisionTypes.hh

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include <egg/math/Vector.hh>
+
+#define KCL_ATTRIBUTE_TYPE(x) ((x)&0x1f)
+#define KCL_TYPE_BIT(x) (1 << (x))
+#define KCL_ATTRIBUTE_TYPE_BIT(x) KCL_TYPE_BIT(KCL_ATTRIBUTE_TYPE(x))
+
+// KCL attribute types
+typedef enum {
```

Comment:
Is there a reason not to use an `enum class` here?

Reply:
I don't see why not. I'll have to add the scope to each use of the enum members

Reply:
I would love to learn an explanation on this behavior

Reply:
> Yes, switching to an enum class actually requires you to define overloads such as `operator&`. Not sure if we want to bother

Good reason. I don't think we want to bother, no.